HackingWithSwift/SwiftUI/Project2/Guess the Flag/Guess the Flag/ContentView.swift

```diff
@@ -7,6 +7,21 @@
 
 import SwiftUI
 
+extension View {
+    
+    @ViewBuilder
+    public func applyIf<Content: View>(
+        _ condition: () -> Bool,
+        @ViewBuilder modifiers: (Self) -> Content
+    ) -> some View {
+        if condition() {
+            modifiers(self)
+        } else {
+            self
+        }
+    }
+}
+
```

Comment:
It’s a good practice to place each extension in a separate file. For naming, consider using a convention like “View+Extensions” for general extensions, or something more specific, such as “View+ConditionalModifiers,” to clarify the purpose of the extension.